core/src/main/scala/cats/mtl/Raise.scala

```diff
@@ -79,6 +79,19 @@ trait Raise[F[_], -E] extends Serializable {
   def ensure[E2 <: E, A](fa: F[A])(error: => E2)(predicate: A => Boolean)(
       implicit A: Monad[F]): F[A] =
     A.flatMap(fa)(a => if (predicate(a)) A.pure(a) else raise(error))
+
+  def fromEither[A](ea: Either[E, A])(implicit F: Applicative[F]): F[A] =
+    ea.fold(raise, F.pure)
+
+  def fromEitherT[E2 <: E, A](ea: EitherT[F, E2, A])(implicit F: Monad[F]): F[A] =
+    F.flatMap(ea.value)(fromEither(_))
+
+  def fromOption[E2 <: E, A](oa: Option[A])(e: => E2)(implicit F: Applicative[F]): F[A] =
+    oa.fold[F[A]](raise(e))(F.pure)
+
+  def fromOptionT[E2 <: E, A](ota: OptionT[F, A])(e: => E2)(implicit F: Monad[F]): F[A] =
+    F.flatMap(ota.value)(oa => fromOption(oa)(e))
+
```

Comment:
```suggestion
  def fromValidated[E2 <: E, A](va: Validated[E2, A])(implicit F: Applicative[F]): F[A] =
    va.fold(raise, F.pure)

```

Reply:
I wonder if we are not accidentally adding a performance penalty comparing with, for example:
```scala
  def fromValidated[E2 <: E, A](va: Validated[E2, A])(implicit F: Applicative[F]): F[A] =
    va match {
      case Validated.Invalid(e) => raise(e)
      case Validated.Valid(a) => F.pure(a)
    }

```

Reply:
Actually we are. Using `fold` requires allocating lambdas, pattern matching does not. I don't think it's important in practice.